content/articles/spring-boot-caching/index.md

```diff
@@ -0,0 +1,332 @@
+		
+
+### Project setup
+We are going to use [spring initializr](https://start.spring.io/) to bootstrap our application.
+		
+
+1. Visit [spring initializr](https://start.spring.io/), input the project name as `SpringBootCaching`.
+2. Add `Spring Web`, `H2 database`, `Spring data JPA`, and `Lombok` as project dependencies.
+3. Click on generate project button to download the project boilerplate code as a zip file.
+4. Extract the zip file and open the uncompressed file in your favorite IDE.
+5. In the `pom.xml` file in the dependencies section, add the dependencies below.
+ ```xml
+<dependencies>
+  <dependency
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-cache</artifactId>
+   </dependency>
+
+   <dependency>
+      <groupId>javax.cache</groupId>
+      <artifactId>cache-api</artifactId>
+   </dependency>
+
+   <dependency>
+      <groupId>org.ehcache</groupId>
+      <artifactId>ehcache</artifactId>
+    </dependency>
+</dependencies>
+``` 
+The above dependencies add `spring-boot-starter-cache`, an abstraction layer implemented by Spring for cache manipulation and `Ehcache` is the cache provider we will be using in our application.
+		
+
+#### Domain layer
+1. Create a new package named `domain` in the root project package.
+2. Create a Java file named `Todo.java` and add the code snippet below.
+		
+```java
+@Entity //Makes this class to be a JPA entity
+@Getter //Creates setters for all the fields in the class
+@Setter //Creates getters for all the fields in the class
+@Builder //Creates a builder pattern for this class
+@NoArgsConstructor //Creates a constructor with no arguements for this class
+@AllArgsConstructor //Creates a constructor with all arguements for this class
+class Todo {
+    @Id // Marks the Id as the primary key
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id", nullable = false)
+    private Long id;
+
+    @Column(name = "title", nullable = false)
+    private String title;
+
+    @Column(name = "description", nullable = false)
+    private String description;
+
+}
+
+```
+In the above code snippet, we create a `Todo` model that will hold an instance of a `todo` in the database.
+		
+
+#### Repository layer
+1. Create a new package named `repository` in the root project package.
+2. Create a new Java interface named `TodoRepository` in the `repository` package created above and add the code snippet below.
+		
+
+```java
+public interface TodoRepository extends JpaRepository<Todo, Long> {
+}
+		
+```
+In the code snippet above, we create a `TodoRepository ` interface that extends the `JpaRepository` interface with all the methods required to perform basic CRUD database operations.
+		
+
+#### Service layer
+1. Create a new package named `service` in the root project directory.
+2. In the `service` package created above, create a new Java interface named `TodoService.java`.
+3. Add the code snippet below in the `TodoService` interface created above.
```

Comment:
Instead of just saying `Add the code snippet below`. Walk the reader through the process of building an application. Explain how, and why we do.

Re-read your article from the perspective of a beginner.

Reply:
Do these for other code snippets too